### DISCUSSION TITLE
Allow CloudWatch Metrics Search Partial Match

### DISCUSSION THREAD

**Discussion by jfharden** [grafana/grafana#71091] [Data Sources]
Using the CloudWatch data source, doing a query using a CloudWatch Metrics Search, if I use the builder it's not possible to perform a [partial match search](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/search-expression-syntax.html#search-expression-partial-match)

If I manually build the search expression I can do it. Take this example:

```
REMOVE_EMPTY(SEARCH('Namespace="AWS/Kinesis" MetricName="IncomingRecords"  StreamName=__nginx_fargate', 'Sum', $__period_auto))
```

Notice the value of the StreamName is not double quoted, this makes it a partial match meaning I can find streams with names such as `test__nginx_fargate` and also `staging__nginx_fargate`.

This example using the builder returns no results:
![Screenshot 2023-07-05 at 16 29 23](https://github.com/grafana/grafana/assets/2170030/9b57e1c4-b6b3-4b2d-b590-534c858964a0)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82599. Feel free to continue the discussion around this there. Thank you!
